quesma/clickhouse/table_discovery.go

```diff
@@ -48,16 +51,18 @@ type tableDiscovery struct {
 	tableDefinitionsLastReloadUnixSec atomic.Int64
 	forceReloadCh                     chan chan<- struct{}
 	ReloadTablesError                 error
+	virtualTableStorage               persistence.JSONDatabase
 }
 
-func NewTableDiscovery(cfg *config.QuesmaConfiguration, dbConnPool *sql.DB) TableDiscovery {
+func NewTableDiscovery(cfg *config.QuesmaConfiguration, dbConnPool *sql.DB, virtualTablesDB persistence.JSONDatabase) TableDiscovery {
 	var tableDefinitions = atomic.Pointer[TableMap]{}
 	tableDefinitions.Store(NewTableMap())
 	result := &tableDiscovery{
-		cfg:              cfg,
-		dbConnPool:       dbConnPool,
-		tableDefinitions: &tableDefinitions,
-		forceReloadCh:    make(chan chan<- struct{}),
+		cfg:                 cfg,
+		dbConnPool:          dbConnPool,
+		tableDefinitions:    &tableDefinitions,
+		forceReloadCh:       make(chan chan<- struct{}),
+		virtualTableStorage: virtualTablesDB,
```

Comment:
Maybe finally, we could merge `tableDefinitions` and `virtualTablesDB` or hide them as common interface